### PR TITLE
chore(swarm): Remove deprecated functions

### PR DESCRIPTION
**Pull request by umgefahren**
## Description

Remove functions deprecated in 0.41.0.

## Links to any relevant issues

Follows from: #3097 and #3119

## Open Questions

None

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by jxs — COMMENTED** [review]
Hi Hannes, looks good on my end! Only the `CHANGELOG.md`'s changes are missing, check [here](https://github.com/libp2p/rust-libp2p/pull/3031/files#diff-78487a5c5ddc3583bba367131aee9230c451649e5f39a2522a784effcf6d6c66 ) for an example. Btw why is clippy complaining?
Thanks!

File: swarm/src/lib.rs
Comment (thomaseizinger) [inline swarm/src/lib.rs:1548]: `with_executor` is unused now I think.

**Review by thomaseizinger — APPROVED** [review]
ACK modulo a changelog entry.

File: swarm/src/lib.rs
Comment (umgefahren) [reply swarm/src/lib.rs:1548]: so let's just delete it then

**Comment by thomaseizinger** [comment]
Before we can merge this, we need to fix the integration test here: https://github.com/libp2p/test-plans/blob/master/ping/rust/src/bin/testplan_0500.rs#L15

Would you mind sending a PR for that?

**Review by thomaseizinger — APPROVED** [review]

**Comment by umgefahren** [comment]
> Before we can merge this, we need to fix the integration test here: https://github.com/libp2p/test-plans/blob/master/ping/rust/src/bin/testplan_0500.rs#L15
> 
> Would you mind sending a PR for that?

On it!

**Review by mxinden — APPROVED** [review]
Thanks @umgefahren for the help here!

**Review by mxinden — COMMENTED** [review]
@umgefahren can you address the CI failures in `libp2p` and `libp2p-gossipsub`?

**Comment by umgefahren** [comment]
Yes will do. Although tomorrow

**Comment by umgefahren** [comment]
> Yes will do. Although tomorrow

Took a little longer, my bad. Should be working now.

**Comment by mxinden** [comment]
Thanks for the follow-ups!

> ACK modulo a changelog entry.

Missing changelog entry @umgefahren. Can you add one?

**Comment by mxinden** [comment]
`libp2p-dcutr` and `libp2p-relay` failures are tracked here: https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1335189994

**Comment by mxinden** [comment]
I forgot to bump the dependents of `libp2p-swarm` :facepalm:. I will do a follow-up pull request tomorrow.

**Comment by mxinden** [comment]
Just to make sure my mental model is correct, cargo semver check could not have detected the above, right? //CC @thomaseizinger

**Comment by thomaseizinger** [comment]
> Just to make sure my mental model is correct, cargo semver check could not have detected the above, right? //CC @thomaseizinger

I think it could but it doesn't today.

cc @obi1kenobi This might be an interesting rule to add:

- crate A exports trait X
- crate B depends on crate A and publicly exports a type implementing trait X
- crate A releases a breaking change (major bump)
- updating the dependency on crate A in crate B is also a breaking change because the trait is a different one

**Comment by obi1kenobi** [comment]
Great idea, thank you! Added to https://github.com/obi1kenobi/cargo-semver-checks/issues/5

**Comment by thomaseizinger** [comment]
> Great idea, thank you! Added to [obi1kenobi/cargo-semver-checks#5](https://github.com/obi1kenobi/cargo-semver-checks/issues/5)

Damn that is a long list! Looking forward to having all those automated some day :)

**Comment by obi1kenobi** [comment]
> Damn that is a long list! Looking forward to having all those automated some day :)

It's not even a complete list, turns out there are lots of ways to break semver in Rust 😅

We just merged a big revamp of how our tests are structured, so we can start knocking out more items from the list pretty soon. Some items are blocked on needing new schema (generics, lifetimes, trait bounds, etc.) but many are not. The ones that aren't blocked are really easy to add: usually 10min or less including writing test cases. If you'd like to try your hand at writing a lint sometime (e.g. ["function becomes `unsafe`"](https://github.com/obi1kenobi/cargo-semver-checks/issues/191)), I'd be happy to help.

**Pull request merged**
